File: android/src/main/java/expo/modules.pagey/PageyModule.kt
package expo.modules.pagey

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PageyModule : Module() {
  override fun definition() = ModuleDefinition {
    name("Pagey")

    function("helloAsync") { options: Map<String, String> ->
      println("Hello 👋")
    }

    viewManager {
      view { context -> 
        PageyView(context) 
      }

      prop("name") { view: PageyView, prop: String ->
        println(prop)
      }
    }
  }
}

File: ios/PageyModule.swift
import ExpoModulesCore

public class PageyModule: Module {
  public func definition() -> ModuleDefinition {
    name("Pagey")

    function("helloAsync") { (options: [String: String]) in
      print("Hello 👋")
      return "Hello 👋"
    }

    viewManager {
      view {
        PageyView()
      }

      prop("numberOfPages") { (view: PageyView, prop: Int) in
        print(prop)
        view.conrol.numberOfPages = prop
      }

      prop("currentPage") { (view: PageyView, prop: Int) in
        print(prop)
        view.conrol.currentPage = prop
      }

      prop("backgroundColor") { (view: PageyView, prop: String) in
        print(prop)
        view.conrol.pageIndicatorTintColor = UIColor(hex: prop)
      }

      prop("currentPageColor") { (view: PageyView, prop: String) in
        print(prop)
        view.conrol.currentPageIndicatorTintColor = UIColor(hex: prop)
      }
    }
  }
}
